src/devices/logitech_g535.c

```diff
@@ -21,21 +86,121 @@ void g535_init(struct device** device)
 
     strncpy(device_g535.device_name, "Logitech G535", sizeof(device_g535.device_name));
 
-    device_g535.capabilities                     = B(CAP_SIDETONE);
+    device_g535.capabilities                     = B(CAP_SIDETONE) | B(CAP_BATTERY_STATUS) | B(CAP_INACTIVE_TIME);
     device_g535.capability_details[CAP_SIDETONE] = (struct capability_detail) { .usagepage = 0xc, .usageid = 0x1, .interface = 3 };
-    device_g535.send_sidetone                    = &g535_send_sidetone;
+    /// TODO: usagepage and id may not be correct for battery status and inactive timer
+    device_g535.capability_details[CAP_BATTERY_STATUS] = (struct capability_detail) { .usagepage = 0xc, .usageid = 0x1, .interface = 3 };
+    device_g535.capability_details[CAP_INACTIVE_TIME]  = (struct capability_detail) { .usagepage = 0xc, .usageid = 0x1, .interface = 3 };
+
+    device_g535.send_sidetone      = &g535_send_sidetone;
+    device_g535.request_battery    = &g535_request_battery;
+    device_g535.send_inactive_time = &g535_send_inactive_time;
 
     *device = &device_g535;
 }
 
 static int g535_send_sidetone(hid_device* device_handle, uint8_t num)
 {
+    int ret = 0;
+
     num = map(num, 0, 128, 0, 100);
 
-    uint8_t set_sidetone_level[MSG_SIZE] = { 0x11, 0xff, 0x04, 0x1e, num };
+    uint8_t buf[HIDPP_LONG_MESSAGE_LENGTH] = { HIDPP_LONG_MESSAGE, HIDPP_DEVICE_RECEIVER, 0x04, 0x1d, num, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
```

Comment:
The 4th byte here changed from `0x1e` to `0x1d`. Don't know why but `0x1d` is the value I saw on my side when setting the sidetone level

Reply:
I will keep that in; we can then see if there are complaints :P